### PR TITLE
Track the current known execution contexts so stuff can use them.

### PR DESCRIPTION
**Pull request by loganfsmyth**

### CODE REVIEW

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 1]
```diff
@@ -254,8 +254,34 @@ function messageCallback(message) {
...
+// inject via evaluatePrivilegd can know what contexts are available.
+const gExecutionContexts = new Map();
+const gContextChangeCallbacks = new Set();
+
 initMessages();
 addEventListener("Runtime.consoleAPICalled", onConsoleAPICall);
+addEventListener("Runtime.executionContextCreated", ({ context }) => {
+  gExecutionContexts.set(context.id, context);
+  for (const callback of gContextChangeCallbacks) {
+    callback(context, "add");
```
Comment (kannanvijayan) [inline third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:267]: Can we wrap all the callback invokes in a try/catch?  It seems like we'd have some other interface for adding things to `gContextChangeCallbacks`, and that code could possibly throw.  Not sure what happens to exceptions at the top level of that script.

**Review by kannanvijayan — APPROVED** [review]
Looks reasonable to me.  If it passes tests it should be fine now.  I added @Domiii as a reviewer too if he wants to take a quick look.

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc (hunk 1 above)
Comment (loganfsmyth) [reply third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:267]: I guess the question there would be what happens if this throws, I guess I was assuming it'd just crash the whole process

Comment (loganfsmyth) [reply third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:267]: But yeah happy to add that.

>  It seems like we'd have some other interface for adding things to gContextChangeCallbacks

Was thinking things would just access the Set directly but happy either way. The main thing with `evaluatePrivileged` is that we haven't at all defined what it should and shouldn't be touching.

**Review by Domiii — APPROVED** [review]
Looks... interesting to me! 😄 

Currently, command errors that are not caught will cause an induced crash of the entire runtime. (They used to only be caught and logged in the past.)

When the Runtime event handler `callback`s fail, our current "crash-on-error" mechanism will not work, so, ideally, something else should be done here.
If we want to enforce the "crash-on-error" rule (which I believe would make a lot of sense, just to be consistent): [this](https://github.com/replayio/chromium-v8/pull/118/files#diff-3373a718e3c6b21f78567214042b719b2e46fba284a43d824c7099ad19ebe6d5R3790) is how it works in the command handler.

cc @klochek @bhackett1024

**Pull request merged**